src/transforms/lua/v2/interop/log.rs

```diff
@@ -0,0 +1,94 @@
+use crate::event::{Event, LogEvent, Value};
+use rlua::prelude::*;
+
+impl<'a> ToLua<'a> for LogEvent {
+    fn to_lua(self, ctx: LuaContext<'a>) -> LuaResult<LuaValue> {
+        ctx.create_table_from(self.into_iter().map(|(k, v)| (k.to_string(), v)))
+            .map(LuaValue::Table)
+    }
+}
+
+impl<'a> FromLua<'a> for LogEvent {
+    fn from_lua(value: LuaValue<'a>, _: LuaContext<'a>) -> LuaResult<Self> {
+        match value {
+            LuaValue::Table(t) => {
+                let mut log = Event::new_empty_log().into_log();
+                for pair in t.pairs() {
+                    let (key, value): (String, Value) = pair?;
+                    log.insert_flat(key, value);
+                }
+                Ok(log)
+            }
+            _ => Err(rlua::Error::FromLuaConversionError {
+                from: "",
```

Comment:
According to https://docs.rs/rlua/0.17.0/rlua/enum.Value.html we can know the value of this type, could we add that here?

Reply:
Yes, actually I even created https://github.com/kyren/rlua/pull/164 to expose an internal method which returns the type as a string which can be used here.